wizard/src/utils/detect_place.ts

```diff
@@ -28,52 +67,75 @@ export class PlaceDetector {
   countryAbbrv3: Set<string>;
   countryNumeric: Set<string>;
 
-  placeTypes: Map<string, string>;
+  // Convenience in-memory maps of types and properties where the keys are their
+  // respective DC Names.
+  placeTypes: Map<string, DCType>;
+  placeProperties: Map<string, DCProperty>;
 
-  static typeFormatMappings = new Map<string, Array<DetectedFormat>>([
-    ["Longitude", [{ propertyName: "longitude", displayName: "Longitude" }]],
-    ["Latitude", [{ propertyName: "latitude", displayName: "Latitude" }]],
-    [
-      "LatLon",
-      [{ propertyName: "GeoCoordinates", displayName: "Geo Coordinates" }],
-    ],
-    [
-      "GeoCoordinates",
-      [{ propertyName: "GeoCoordinates", displayName: "Geo Coordinates" }],
-    ],
+  // A set of all place types and their associated properties.
+  placeTypesAndProperties: Set<TypeProperty>;
+
+  // Mapping between Place types and supported properties associated with each
+  // type.
+  static keyTypePropertyMappings = new Map<string, Array<TPName>>([
+    ["longitude", [{ tName: "GeoCoordinates", pName: "longitude" }]],
```

Comment:
curious what the difference between the key of the keyTypePropertyMappings map and the tName values is because they've both been described as place type?

Reply:
Good question and I should update the comment above. The difference is that the keys in the keyTypePropertyMappings are the (approx) column names that we might anticipate a user would put in their csv. So, when detecting if a particular column name matches a valid location type, we can't rely on just matching the DC Names. The keys here are what we expect to match with a user's csv headers. If there is a match, then the values in keyTypePropertyMappings are the DC Type and DC Property we can infer.